### PR TITLE
Various John commits

### PR DESCRIPTION
**Pull request by muschellij2**
Easier to maintain scraping code for regions.  Probably an API call makes more sense, but this is how I got the regions anyway, so figured I'd make it simpler and updatable.  Also, realized region_set didn't work unless same choices were available.

### CODE REVIEW

**Comment by MarkEdmondson1234** [comment]
Good stuff :) From a quick Google I think this API endpoint would be the one to use for an API call solution https://cloud.google.com/run/docs/reference/rest/v1/projects.locations/list - have put it in an issue to be tackled later https://github.com/MarkEdmondson1234/googleCloudRunner/issues/123

**Comment by muschellij2** [comment]
Crud, I added a fix in here for #124 in here as well.  PR is polluted, so let me konw how you want to break if you don't want both aspects.

**Comment by MarkEdmondson1234** [comment]
No worries, I'll merge them both in.  Ideally a test (Even if it only works locally) would also be included for the two issues - they will be run with the test credentials then when I merge into master.

**Comment by muschellij2** [comment]
I also just added the argument `binary_mode` for `cr_buildstep_secret` as I had a `token.rds` in there and it doesn't seem to like that unless you get the payload data and base64 decrypt it as per https://cloud.google.com/secret-manager/docs/creating-and-accessing-secrets#a_note_on_resource_consistency

It could be the default without the argument, but I'm unsure if that will screw up any plain text secrets, so figured it'd be better to not make breaking changes.

So I need:
1) Test for different service account (I think I need a name of an account other than the default runner account, unless the test should fail)
2) A test for a binary secret - I think you need to make one with a binary file and then I can use a test to see if it can be read accurately - I think just make a data.frame into an .rds (compressed?) and we can use that as a simple test.

**Comment by muschellij2** [comment]
This allows a different service account to be run as per: https://cloud.google.com/build/docs/securing-builds/configure-user-specified-service-accounts

File: inst/ssh/github_setup.sh [hunk 1]
```diff
@@ -1,10 +1,10 @@
-chmod 600 /root/.ssh/id_rsa
+chmod 600 /root/.ssh/id_ed25519
```
Comment (MarkEdmondson1234) [inline inst/ssh/github_setup.sh:1]: What was the reason for this change?

File: data-raw/cr_regions.R [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+             "us-east1")
+cr_regions <- c(cr_regions,
+             "asia-east1", "asia-northeast1", "asia-northeast2", "europe-north1",
+             "europe-west1", "europe-west4", "us-central1", "us-east1", "us-east4",
+             "us-west1", "asia-east2", "asia-northeast3", "asia-southeast1",
+             "asia-southeast2 ", "asia-south1", "asia-south2", "australia-southeast1",
+             "australia-southeast2", "europe-central2", "europe-west2", "europe-west3",
+             "europe-west6", "northamerica-northeast1", "northamerica-northeast2",
+             "southamerica-east1", "us-west2", "us-west3", "us-west4")
+cr_regions <- unique(cr_regions)
```
Comment (MarkEdmondson1234) [inline data-raw/cr_regions.R:24]: This file generation script can be removed from this pull now as its in via https://github.com/MarkEdmondson1234/googleCloudRunner/commit/3effcee84f94926e46f93a32f6bbbee408a4d96c

File: inst/ssh/github_setup.sh (hunk 1 above)
Comment (muschellij2) [reply inst/ssh/github_setup.sh]: I think this isn't actually relevant.  I thought it'd be good to name the identity file corerctly depending on the encryption (RSA vs. ED25519), but I don't think it's necessary.

Comment (muschellij2) [reply inst/ssh/github_setup.sh]: I can remove this addition if you want, unless you think there is any benefit to having a different key option?

File: data-raw/cr_regions.R (hunk 2 above)
Comment (MarkEdmondson1234) [reply data-raw/cr_regions.R:24]: The list will get checked each time the website is built, I hitched a ride on the back of the website commit bot https://github.com/MarkEdmondson1234/googleCloudRunner/blob/45ba339fc9865c96b79a7e6ab3a2e4e58242e824/inst/cloudbuild/cloudbuild_pkgdown.yml#L51

**Comment by MarkEdmondson1234** [comment]
@muschellij2 I would like to merge but can't in the browser as its flagging the .html files as having too complex changes to resolve.  I know these are unimportant changes, so if you can put in a commit that will remove the doc changes you made I can merge it in. 

Also the linter or something added a lot of whitespace changes and " to ' etc. which made it harder to parse what was important so for next time it would be cool to put linter changes in another pull request.

**Comment by muschellij2** [comment]
OK - should be ready to go.

**Comment by MarkEdmondson1234** [comment]
Done!

**Pull request merged**